packages/edit-site/src/components/global-styles/dimensions-panel.js

```diff
@@ -14,31 +14,34 @@ import { __experimentalUseCustomSides as useCustomSides } from '@wordpress/block
 /**
  * Internal dependencies
  */
-import { useSetting } from '../editor/utils';
+import { getSupportedGlobalStylesPanels, useSetting, useStyle } from './hooks';
 
 const AXIAL_SIDES = [ 'horizontal', 'vertical' ];
 
-export function useHasDimensionsPanel( context ) {
-	const hasPadding = useHasPadding( context );
-	const hasMargin = useHasMargin( context );
-	const hasGap = useHasGap( context );
+export function useHasDimensionsPanel( name ) {
+	const hasPadding = useHasPadding( name );
+	const hasMargin = useHasMargin( name );
+	const hasGap = useHasGap( name );
 
 	return hasPadding || hasMargin || hasGap;
 }
 
-function useHasPadding( { name, supports } ) {
+function useHasPadding( name ) {
+	const supports = getSupportedGlobalStylesPanels( name );
 	const settings = useSetting( 'spacing.customPadding', name );
```

Comment:
All these `useSetting` calls need to pull the value (either `useSetting( 'typography...', name)[ 0 ]` or `[ value ] = useSetting('typography...', name )`), correct?